tools/merge-databases/merge-databases.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# The purpose of this script is a one time data migration from
+# builder_originsrv and builder_sessionsrv -> builder
+
+set -euo pipefail
+
+usage() {
+  echo "Usage: merge-databases.sh"
+  exit 1
+}
+
+if [ -z "${PGPASSWORD:-}" ]; then
+  echo "Please ensure that PGPASSWORD is exported in your environment and run this script again."
+  exit 1
+fi
+
+echo "This script will take all of the data that's currently in builder_originsrv and builder_sessionsrv and merge them into builder."
+echo "Please ensure that you have completed a full database backup before running this script."
+echo
+
+read -rp "Do you want to proceed? [y/n] " answer
+clean_answer=$(echo "$answer" | xargs | tr '[:upper:]' '[:lower:]')
+
+if [ "${clean_answer::1}" != "y" ]; then
+  echo "Aborting"
+  exit 1
+fi
+
+execute_sql() {
+  local database="$1"
+  local sql="$2"
+  local port=${PORT:-5432}
+
+  hab pkg exec core/postgresql psql -t -U hab -h 127.0.0.1 -p "${port}" -c "${sql}" -d "${database}"
+}
+
+# before we do anything, let's check to see if this has already been done
+echo "Checking to see if the shard migration has already happened"
+if ! execute_sql "builder_originsrv" "SELECT shard_migration_complete FROM flags;" | grep -q 't'; then
+echo "The flags table is not present in the database, which is required for this script to run."
+echo "It's possible something is amiss with your Builder database migrations."
+exit 1
+fi
+
+# before we do anything part 2, let's check to see if this has already been done
+echo "Checking to see if the db migration has already happened"
+# shellcheck disable=SC2046
```

Comment:
Why not
```bash
if [ "$(execute_sql "builder" "SELECT count(*) from origins;")" -gt 0 ]; then
```
?

Reply:
I'm pretty sure that breaks the inner quoting and I can't escape the inner quotes because then it translates into literal sql

Reply:
It doesn't break the inner quoting. Shell handles this fairly differently from other languages in that you can nest quotes and expansions to an arbitrary depth. For example:
```bash
jbauman@ubuntu:~/builder$ touch "foo bar"
jbauman@ubuntu:~/builder$ echo "$(ls -l "foo bar")"
-rw-rw-r-- 1 jbauman jbauman 0 Aug 29 08:14 foo bar
```